tests/api_resources/abstract/test_api_resource.py

```diff
@@ -101,3 +113,65 @@ def test_raise_on_incorrect_id_type(self):
         for obj in [None, 1, 3.14, dict(), list(), set(), tuple(), object()]:
             with pytest.raises(stripe.error.InvalidRequestError):
                 self.MyResource.retrieve(obj)
+
+    @pytest.fixture
+    def resource(self):
+        return self.MyResource.construct_from(
+            {"id": "foo"},
+            key="newkey",
+            stripe_version="2023-01-01",
+            stripe_account="acct_foo",
+        )
+
+    @pytest.fixture
+    def check_request_options(self, http_client_mock):
+        def check_request_options(
+            *, api_key=None, stripe_version=None, stripe_account=None
+        ):
+            extra_headers = {}
+            if stripe_account is None:
+                extra_headers = {"Stripe-Account": None}
+            if api_key is None:
+                api_key = stripe.api_key
+            if stripe_version is None:
+                stripe_version = stripe.api_version
+
+            http_client_mock.assert_requested(
+                "get",
+                path="/v1/myresources/foo",
+                api_key=api_key,
+                stripe_version=stripe_version,
+                extra_headers=extra_headers,
+            )
+
+        return check_request_options
+
+    def test_class_method_does_not_forward_options(
```

Comment:
Do we need a scenario where parameters are passed into the method call? `Account.create(, stripe_version = ...)

Reply:
Good call, I'll add that